src/utils/drawing.js

```diff
@@ -0,0 +1,207 @@
+import { LINE_LEN } from '../constants';
+import { rgb } from 'pdf-lib';
+
+/**
+ * @typedef Point
+ * @type {object}
+ * @property {number} x - horizontal position
+ * @property {number} y - vertical position
+ */
+
+/**
+ * @typedef Line
+ * @type {object}
+ * @property {Point} start - start position
+ * @property {Point} end - end position
+ * @property {number} [opacity] - line opacity
+ * @property {number[]} [dashArray] - sequence of dash and gap lengths to be repeated for a dashed line
+ */
+
+/**
+ *  @param {boolean} side2flag - whether we're on the back or not.
+ * @param {boolean} duplexrotate - if alternate sides are rotated or not
+ * @param {number[]} papersize - paper dimensions
+ * @param {number} per_sheet - pages per sheet of paper
+ * @returns {Line[]}
+ */
+export function drawFoldlines(side2flag, duplexrotate, papersize, per_sheet) {
```

Comment:
this is a personal nit and probably just a programming langue issue, but I would wish for an object here with named values rather than a series of parameters that can make future modification or extension less clear. My very mild $0.02 that you are very welcome to ignore.

Reply:
I think the config object strategy works for stuff with a lot of params, but I do generally find it slightly less readable for short params lists, so I'm gonna leave this one as-is for now.

(some of the config-object stuff would also probably be better served by making them class methods on smaller classes, and then using class properties, but that's farther down the line, lol)